sos monitoring script

gkaratha opened the pull request:
Script to monitor the HLT_DoubleMu3_DZ_PFMET50_PFMHT60. This path is used by the soft opposite sign analysis for SUSY searches with small mass difference. It uses the SOSMonitor to measure efficiency of both legs (MET and muon)

gkaratha committed:
sos monitoring script

cmsbuild commented:
A new Pull Request was created by  @gkaratha  for CMSSW_9_2_0_patchX.

It involves the following packages:

DQMOffline/Trigger


@kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.
Note that this branch is designed for requested bug fixes specific to the CMSSW_9_2_0 release.
If you wish to make a pull request for the CMSSW_9_2_X release cycle, please use the CMSSW_9_2_X branch instead

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mtosi commented on DQMOffline/Trigger/plugins/SOSMonitor.cc:
```diff
@@ -0,0 +1,316 @@
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DQM/TrackingMonitor/interface/GetLumi.h"
```
drop it

mtosi commented on DQMOffline/Trigger/plugins/SOSMonitor.cc:
```diff
@@ -0,0 +1,316 @@
+
+#include "TLorentzVector.h"
+
+double sosMAX_PHI = 3.2;
```
add them to the class as members

mtosi commented on DQMOffline/Trigger/plugins/SOSMonitor.cc:
```diff
@@ -0,0 +1,316 @@
+  , jetSelection_ ( iConfig.getParameter<std::string>("jetSelection") )
+  , eleSelection_ ( iConfig.getParameter<std::string>("eleSelection") )
+  , muoSelection_ ( iConfig.getParameter<std::string>("muoSelection") )
+  , njets_      ( iConfig.getParameter<int>("njets" )      )
```
change in unsigned

mtosi commented on DQMOffline/Trigger/plugins/SOSMonitor.h:
```diff
@@ -0,0 +1,117 @@
+  MON sosHTMON_;
+
+
+  GenericTriggerEventFlag* num_genTriggerEventFlag_;
```
unique_ptr

mtosi commented on DQMOffline/Trigger/plugins/SOSMonitor.cc:
```diff
@@ -0,0 +1,316 @@
+  , mu2_pt_cut_ (iConfig.getParameter<int>("mu2_pt_cut"))
+{
+
+  sosMON_.numerator   = nullptr;
```
do you really need it ?
otherwise, drop it

mtosi commented on DQMOffline/Trigger/plugins/SOSMonitor.cc:
```diff
@@ -0,0 +1,316 @@
+  
+  float met = pfmet.pt();
+  float phi = pfmet.phi();
+  int jet60=0;
```
ok, I see that all comments I made in your branch have not been addressed yet
please, commits the needed changes
then, I'll continue the review

mtosi commented:
tracked by https://github.com/cms-sw/cmssw/issues/19142

rcastello commented:
In case you missed, a new PR has been circulated at https://github.com/cms-sw/cmssw/pull/19337 including the requested changes. Thanks

gkaratha commented:
Dear Mia, we addressed your comments here: https://github.com/cms-sw/cmssw/pull/19337 . I am not 100% sure if everyone is informed automatically so i posted here as well. Thanks

mtosi commented:
ciao
i'm in holiday
i've just added some comments in the other PR
if this is no longer needed, please close

dmitrijus commented:
-1 superseeded

fwyzard commented:
superseded by which PR ?
